### DISCUSSION TITLE
Terraform for AWS ECS and Neptune

### DISCUSSION THREAD

**Discussion by BeyondEvil** [aws/graph-explorer#224] [Show and tell]
Since sharing is caring, I wanted to share the Terraform we used to set up Graph Explorer and connect it to Neptune.

[example repo](https://github.com/BeyondEvil/graph-explorer-example)

**Comment by triggan** [comment]
Just out of curiosity, is there a reason why you all are choosing to use Graph Explorer from ECS versus the Graph Explorer that gets installed via a Neptune Notebook instance?

**Reply by BeyondEvil** [reply]
I’m an infra engineer with zero knowledge of Notebooks. I did this is a PoC for work, so I used what I know.

**Reply by qwaider** [reply]
In our case, we wanted to have GE for the users to visualise as web-page without logging to notebook. also maintaining notebook is more expensive than an EC2.

We are yet struggling connecting GE proxy server and GE ui using a nginx reverse proxy... but we are working on it.

**Reply by kmcginnes** [reply]
@qwaider I have no idea if this will help you, but one thing that it took me too long to realize is that the Graph Explorer UI is actually hosted by the proxy server itself when deployed as a Docker image in production mode.

https://github.com/aws/graph-explorer/blob/ede994b44a465f0310de7ffc3f0b52d6db1c89ba/packages/graph-explorer-proxy-server/node-server.js#L177-L186

I haven't had time to experiment with this idea, but I wonder if issues arise (i.e. CORS) if you host the UI and proxy as separate servers. It probably shouldn't be this way, but I wouldn't be surprised if it is, currently.

**Reply by qwaider** [reply]
Actually, It is the CORS.
So both are in the same server. If I use the server ip... both servers are working fine..
If I use the URL using the nginx reverse server then UI can't reach the proxy server .. neither by ip nor by the url.

**Reply by kmcginnes** [reply]
That is not a scenario that I've ever tried. So I can't offer any specific help.

However, there is an environment value that is used during the Docker entry that builds the SSL certificate using the server's public URL. Perhaps following that thread through the code could shed some light on what is preventing your situation from working.

https://github.com/aws/graph-explorer/blob/main/additionaldocs/development.md#host
